dbt/adapters/snowflake/relation_configs/catalog.py

```diff
@@ -0,0 +1,121 @@
+    https://docs.snowflake.com/en/sql-reference/sql/create-dynamic-table#create-dynamic-iceberg-table
+
+    The following parameters are configurable by dbt:
+    - table_format: format for interfacing with the table, e.g. default, iceberg
+    - external_volume: name of the external volume in Snowflake
+    - base_location: the directory within the external volume that contains the data
+        *Note*: This directory can’t be changed after you create a table.
+
+    The following parameters are not currently configurable by dbt:
+    - name: snowflake
+    """
+
+    table_format: Optional[TableFormat] = TableFormat.default()
+    name: Optional[str] = "SNOWFLAKE"
+    external_volume: Optional[str] = None
+    base_location: Optional[str] = None
+
+    @property
+    def validation_rules(self) -> Set[RelationConfigValidationRule]:
+        return {
+            RelationConfigValidationRule(
+                (self.table_format == "default")
+                or (self.table_format == "iceberg" and self.base_location is not None),
+                DbtConfigError("Please provide a `base_location` when using iceberg"),
+            ),
+            RelationConfigValidationRule(
+                (self.table_format == "default")
+                or (self.table_format == "iceberg" and self.name == "SNOWFLAKE"),
+                DbtConfigError(
+                    "Only Snowflake catalogs are currently supported when using iceberg"
+                ),
+            ),
+        }
+
+    @classmethod
+    def from_dict(cls, config_dict: Dict[str, Any]) -> Self:
+        kwargs_dict = {
+            "name": config_dict.get("name"),
+            "external_volume": config_dict.get("external_volume"),
+            "base_location": config_dict.get("base_location"),
+        }
+        if table_format := config_dict.get("table_format"):
+            kwargs_dict.update({"table_format": TableFormat(table_format)})
+        return super().from_dict(kwargs_dict)
+
+    @classmethod
+    def parse_relation_config(cls, relation_config: RelationConfig) -> Dict[str, Any]:
+
+        if relation_config.config.extra.get("table_format") is None:
+            return {"table_format": "default"}
+
+        config_dict = {
+            "table_format": relation_config.config.extra.get("table_format"),
+            "name": "SNOWFLAKE",  # this is not currently configurable
+        }
+
+        if external_volume := relation_config.config.extra.get("external_volume"):
+            config_dict.update({"external_volume": external_volume})
+
+        if base_location := relation_config.config.extra.get("base_location_subpath"):
+            config_dict.update({"base_location": base_location})
+
+        return config_dict
+
+    @classmethod
+    def parse_relation_results(cls, relation_results: RelationResults) -> Dict[str, Any]:
+        catalog_results: "agate.Table" = relation_results["catalog"]
+
+        if catalog_results is None or len(catalog_results) == 0:
```

Comment:
Python tends to like 

`if not catalog_results` since empty collections are falsey

over

`if catalog_results is None or len(catalog_results) == 0:`

Are the rules different for Agate tables?

Reply:
Good catch. This is just an artifact from iteration. I think I originally had `if catalog_results is None or catalog.<something>` and I needed to do the two part check to be able to check the attribution in the second. But then I decided to always return a catalog and just didn't update this. I'll fix it.